code/api/tests/test_posts.py

```diff
@@ -100,4 +102,75 @@ def test_get_posts_paginated_page_2(self):
         
         response = self.client.get(reverse('api:posts', args=(post1.author.id,)) + f'?page={page}&size={size}')
         self.assertEqual(response.status_code, 200)
-        self.assertJSONEqual(response.content, expected)
\ No newline at end of file
+        self.assertJSONEqual(response.content, expected)
+
+
+class PostViewTest(TestCase):
+
+    basicAuthHeaders = {
+        'HTTP_AUTHORIZATION': 'Basic %s' % base64.b64encode(b'remotegroup:topsecret!').decode("ascii"),
+    }
+
+    def setUp(self):
+        Node.objects.create(host=HOST, basic_auth_credentials='remotegroup:topsecret!')
+
+    def test_get_single_post(self):
+        author = mixer.blend(LocalAuthor)
+        post = create_post("single-post", author)
+
+        expected = get_post_json(post)
+        response = self.client.get(
+            reverse('api:post', args=(author.id, post.id)),
+            HTTP_REFERER=HOST,
+            **self.basicAuthHeaders
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(response.content, expected)
+
+    def test_get_post_notfound_1(self):
+        author1 = mixer.blend(LocalAuthor)
+        author2 = mixer.blend(LocalAuthor)
+        post = create_post("single-post", author2)
+
+        response = self.client.get(
+            reverse('api:post', args=(author1.id, post.id)),
+            HTTP_REFERER=HOST,
+            **self.basicAuthHeaders
+        )
+        self.assertEqual(response.status_code, 404)
+
+    def test_get_post_notfound_2(self):
+        author = mixer.blend(LocalAuthor)
+        post = create_post("single-post", author)
+        post.id = 100
```

Comment:
Seems that this line duplicates the post. Idk why, just weird Django stuff I guess. The test does pass, but I imagine that's not what you intended. I think you'll have to set the id when you create the post.

Reply:
I see, that's really good catch. I guess it doesn't allow for the pk to change, but instead creates a new post with the new pk (and keeps the previous one). I'll set the id when creating the post then. Thank you! ♥